pkg/util/url.go

```diff
@@ -0,0 +1,15 @@
+package util
+
+import (
+	"fmt"
+	"net/url"
+)
+
+func ParseURL(urlStr string) (*url.URL, error) {
+	u, err := url.Parse(urlStr)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing URL. Ensure that any embedded values such as username & password have been URL encoded: %w", err)
+	}
+
+	return u, nil
```

Comment:
Just use errors.Wrap from `github.com/pkg/errors`. It will return nil if the passed-in error is nil.

```suggestion
	u, err := url.Parse(urlStr)
	return u, errors.Wrap(err, "ensure that DSN parameters such as usernames or passwords containing special characters have been URL encoded")
	}

	return u, nil
```

Reply:
Done